### PR TITLE
Mavetization

### PR DESCRIPTION
**Pull request by javydreamercsw**
Sumarizing:

- Moved non java source files to the resources folder so they find their way into the jars as needed.

- Moved sources (unchanged) to src/java/main.

- Maven updates the libs folder and download licences there as well. (Since the changes in that folder. Just using the ones from maven)

- Updated build.ant.properties so it finds the new libraries (without versions in filename).

### CODE REVIEW

**Comment by javydreamercsw** [comment]
There's a test failing (json) which I can't explain. I'm using the same json library version.

**Comment by javydreamercsw** [comment]
Weird, I get one failure when I run it on my Ubuntu box:

testWriteToJson(marauroa.common.game.RPObjectTest)  Time elapsed: 0.013 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: "\"c\":\"\",\"a\":{\"mykey\":\"myvalue\"},\"s\":{\"myslot\":[{\"c\":\"\",\"a\":{\"id\":\"0\"}},{\"c\":\"\",\"a\":{\"id\":\"1\",\"mychildkey\":\"mychildvalue\"}}]}"
     but: was "\"c\":\"\",\"a\":{\"mykey\":\"myvalue\"},\"s\":{\"myslot\":[{\"c\":\"\",\"a\":{\"id\":\"0\"}},{\"c\":\"\",\"a\":{\"mychildkey\":\"mychildvalue\",\"id\":\"1\"}}]}"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at marauroa.common.game.RPObjectTest.testWriteToJson(RPObjectTest.java:427)

**Comment by nhnb** [comment]
This PR looks a lot cleaner, thank you.

On a quick glance, I think that is a bug in the test itself. It seems to depend on String.hashCode(), which returns different values depending on the Java version. So we can ignore that test for the sake of this PR.

There is another issue, however. Eclipse fails to compile the project, listing those errors:

- Archive for required library: 'libs/h2.jar' in project 'marauroa' cannot be read or is not a valid ZIP file	
- Archive for required library: 'libs/log4j.jar' in project 'marauroa' cannot be read or is not a valid ZIP file	

Unix "file" commands reports both files as "Zip archive data, at least v2.0 to extract", all other files are reported as "Zip archive data, at least v1.0 to extract" or "Java archive data (JAR)"

ant dist results in a similar error message:
[javac] error: error reading /home/hendrik/workspace/marauroa/libs/log4j.jar; error in opening zip file

I am using javac 1.8.0_121, so this is not caused by an outdated java version.



I am not sure, how to setup the source folders correctly.  The new location of src/java/main is a subfolder of the old src folder. But the js subfolder is still in src. In src/main/resources are both files that should go into the binary, and files that are only used in tests. In other maven projects, "test" seems to be below src as well, next to main.

**Comment by balaur64** [comment]
* I still have issues with missing Java Web Start (see also the discussion on [the initial patch](https://sourceforge.net/p/arianne/patches/646/)):
```
[ERROR] Failed to execute goal on project Marauroa: Could not resolve dependencies for project net.sourceforge.javydreamercsw:Marauroa:jar:3.9.5-SNAPSHOT: Could not find artifact javax.jnlp:jnlp-api:jar:7.0 at specified path /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/javaws.jar -> [Help 1]
```
* Why don't we just let Maven download the dependencies, instead of keeping them in `libs/`?
* @nhnb there's a bug in the recent maven eclipse plug-ins, that sometimes causes incorrect dowloads of dependencies. Maybe you could try to delete the affected jars, then run maven from the command-line.

**Comment by javydreamercsw** [comment]
Right now the libs folder is maintained by Maven. I have it there if we want to keep the ant support for compatibility.

Java Web start is being retrieved from the local Java installation. I assume you are not using Oracle Java.

**Comment by balaur64** [comment]
In my opinion we should move to Maven completely and not maintain 2 build systems.
However, I'm not the best person to decide this, and for the moment it should be OK the way you did it.
Anyway, if we would decide to keep Ant, then maybe we could somehow make use of Ivy for the dependencies.

Regarding javaws, yes - I am not using Oracle Java. I sent you a [PR](https://github.com/javydreamercsw/marauroa/pull/1).

**Comment by javydreamercsw** [comment]
Merged that PR.  I agree with maintaining only one build system, but that seems like a decision for the higher ups.

**Comment by javydreamercsw** [comment]
Fixed the conflicts in case this is still of interest.

File: src/main/java/marauroa/common/net/MessageFactory.java [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      Message tmp = null;
+      try
+      {
+        Class<?> messageType = factoryArray.get(messageTypeIndex);
+        tmp = (Message) messageType.newInstance();
+        tmp.setProtocolVersion(networkProtocolVersion);
+        ByteArrayInputStream in = new ByteArrayInputStream(data);
+        if (offset > 0)
+        {
+          in.skip(offset);
```
[the remark this answers is not in the document]
Comment (javydreamercsw) [reply src/main/java/marauroa/common/net/MessageFactory.java]: This has been so long ago I don't even remember but this doesn't seem like a change I did. I only moved stuff to normal Maven structure.

**Comment by javydreamercsw** [comment]
Any updates?

**Pull request closed without merging**